implementations/rust/ockam/ockam_command/src/secure_channel/listener/create.rs

```diff
@@ -33,42 +34,33 @@ pub struct SecureChannelListenerNodeOpts {
 }
 
 impl CreateCommand {
-    pub fn run(self, options: CommandGlobalOpts) {
-        let cfg = options.config;
-        let node = extract_address_value(&self.node_opts.at).unwrap_or_else(|_| "".to_string());
-        let port = cfg.get_node_port(&node).unwrap();
-
-        connect_to(port, self, |ctx, cmd, rte| async {
-            create_listener(&ctx, cmd.address, cmd.authorized_identifier, rte).await?;
-            drop(ctx);
-            Ok(())
-        });
+    pub fn run(self, opts: CommandGlobalOpts) {
+        node_rpc(rpc, (opts, self));
     }
 }
 
-pub async fn create_listener(
-    ctx: &ockam::Context,
-    addr: Address,
-    authorized_identifiers: Option<Vec<IdentityIdentifier>>,
-    mut base_route: Route,
-) -> anyhow::Result<()> {
-    let resp: Vec<u8> = ctx
-        .send_and_receive(
-            base_route.modify().append(NODEMANAGER_ADDR),
-            api::create_secure_channel_listener(&addr, authorized_identifiers)?,
-        )
-        .await?;
-
-    let response = api::parse_create_secure_channel_listener_response(&resp)?;
+async fn rpc(ctx: Context, (opts, cmd): (CommandGlobalOpts, CreateCommand)) -> crate::Result<()> {
+    run_impl(&ctx, (opts, cmd)).await
+}
 
-    match response.status() {
-        Some(Status::Ok) => {
-            println!("/service/{}", addr.address());
-            Ok(())
-        }
-        _ => {
-            eprintln!("An error occurred while creating secure channel listener",);
-            std::process::exit(exitcode::CANTCREAT)
-        }
-    }
+async fn run_impl(
+    ctx: &Context,
```

Comment:
this context was [mutable at one point](https://github.com/build-trust/ockam/blob/042f5797fb51383fef7dbe1d0cc007d896aa5113/implementations/rust/ockam/ockam_command/src/secure_channel/listener/create.rs#L56) as well as the one at secure-channel listener list [still is mutable](https://github.com/build-trust/ockam/blob/5b8ffc4051d6fee08354bc5a5f26abe8ab05ad3e/implementations/rust/ockam/ockam_command/src/secure_channel/listener/list.rs#L31), i wonder what's the difference?

Reply:
When using the `rpc` abstraction, a mutable context is not needed. Actually, the `mut` reference is not being used in the `secure-channel listener list` command and all would be still the same with a referenced context.

We used to borrow the context to stop it explicitly at the end of every command. That's now done within the `node_rpc` function automatically.